modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main/categorization/vocabularies/EditVocabulary.tsx

```diff
@@ -0,0 +1,210 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayIcon from '@clayui/icon';
+import ClayLayout from '@clayui/layout';
+import {ClayVerticalNav} from '@clayui/nav';
+import {ManagementToolbar} from 'frontend-js-components-web';
+import {navigate, sub} from 'frontend-js-web';
+import React, {useState} from 'react';
+
+import VocabularyService from '../services/VocabularyService';
+import {AssetType} from '../types/AssetType';
+import {IVocabulary} from '../types/IVocabulary';
+import EditAssociatedAssetTypes from './EditAssociatedAssetTypes';
+import EditGeneralInfo from './EditGeneralInfo';
+
+export default function EditVocabulary({
+	assetTypes,
+	backURL,
+	defaultLanguageId,
+	locales,
+	siteId,
+	spritemap,
+	vocabulary,
+}: {
+	assetTypes: AssetType[];
+	backURL: string;
+	defaultLanguageId: string;
+	locales: any[];
+	siteId: number;
+	spritemap: string;
+	vocabulary: IVocabulary;
+}) {
+	const [activeVerticalNavKey, setActiveVerticalNavKey] = useState('general');
+	const [initialItemData, setInitialItemData] = useState<IVocabulary>(
+		vocabulary
+			? vocabulary
+			: {
+					description: '',
+					name: '',
+					name_i18n: {
+						[defaultLanguageId]: '',
+					},
+				}
+	);
```

Comment:
Conventionally, you can have the prop instead be called `initialVocabulary` and then have the state as `[vocabulary, setVocabulary]`. Those can then be passed into `EditVocabulary` as `vocabulary` and `onChangeVocabulary`.